titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/apigateway/DefaultApiGatewayCallbackService.java

```diff
@@ -0,0 +1,108 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+
+@Singleton
+public class DefaultApiGatewayCallbackService implements ApiGatewayCallbackService {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultApiGatewayCallbackService.class);
+    private JobServiceGateway jobServiceGateway;
+    private static final String DIMENSION_NAME = "custom-resource:ResourceType:Property";
+
+    @Inject
+    public DefaultApiGatewayCallbackService(JobServiceGateway jobServiceGateway) {
+        this.jobServiceGateway = jobServiceGateway;
+    }
+
+    @Override
+    public Observable<ScalingPayload> getJobInstances(String jobId) {
+        TaskQuery taskQuery = TaskQuery.newBuilder()
+                .putFilteringCriteria("jobIds", jobId)
+                .putFilteringCriteria("taskStates", "Started")
+                .setPage(Page.newBuilder().setPageSize(1).build()).build();
+
+        return jobServiceGateway.findTasks(taskQuery)
+                .map(taskQueryResult -> taskQueryResult.getPagination().getTotalItems())
+                .flatMap(numStartedTasks -> jobServiceGateway.findJob(jobId).map(job -> Pair.of(job, numStartedTasks)))
+                .flatMap(jobTasksPair -> {
+                    Job job = jobTasksPair.getLeft();
+                    if (!job.getJobDescriptor().hasService()) {
+                        return Observable.error(JobManagerException.notServiceJob(jobId));
+                    }
+                    ServiceJobSpec jobSpec = job.getJobDescriptor().getService();
+
+                    ScalingPayload scalingPayload = new ScalingPayload();
+                    scalingPayload.setActualCapacity(jobTasksPair.getRight());
+                    scalingPayload.setDesiredCapacity(jobSpec.getCapacity().getDesired());
+                    scalingPayload.setDimensionName(DIMENSION_NAME);
+                    scalingPayload.setResourceName(jobId);
+                    scalingPayload.setScalableTargetDimensionId(jobId);
+                    scalingPayload.setVersion(getAcceptedTimestamp(job));
+                    return Observable.just(scalingPayload);
+                });
+    }
+
+    @Override
+    public Observable<ScalingPayload> setJobInstances(String jobId, ScalingPayload scalingPayload) {
+        JobCapacityWithOptionalAttributes jobCapacityWithOptionalAttributes = JobCapacityWithOptionalAttributes.newBuilder()
+                .setDesired(UInt32Value.newBuilder().setValue(scalingPayload.getDesiredCapacity()).build()).build();
+        JobCapacityUpdateWithOptionalAttributes jobCapacityRequest = JobCapacityUpdateWithOptionalAttributes.newBuilder()
+                .setJobId(jobId)
+                .setJobCapacityWithOptionalAttributes(jobCapacityWithOptionalAttributes).build();
+        return jobServiceGateway.updateJobCapacityWithOptionalAttributes(jobCapacityRequest)
+                .andThen(getJobInstances(jobId));
+    }
+
+    private String getAcceptedTimestamp(Job job) {
```

Comment:
I believe there is a utility method somewhere that can do this for any given state. It might need to be moved to the common module though.